service-clients/src/main/java/com/google/cloud/pso/rag/common/JDBCHelper.java

```diff
@@ -0,0 +1,92 @@
+package com.google.cloud.pso.rag.common;
+
+import com.pgvector.PGvector;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class JDBCHelper {
+  private static final ExecutorService databaseExecutor = Executors.newFixedThreadPool(1);
+
+  public static CompletableFuture<ResultSet> executeSQLStmtAsync(
+      String sql, String jdbcUrl, String user, String password) {
+    return CompletableFuture.supplyAsync(
+        () -> {
+          // JDBC operations happen in a thread from databaseExecutor
+          try (Connection connection = DriverManager.getConnection(jdbcUrl, user, password);
+              Statement stmt = connection.createStatement()) {
```

Comment:
we should use prepared statements here, let the driver take care of the decisions on how to deal with caches, param validation and other things. 

you can pass a lambda to set the params on a per request basis.